clippy_lints/src/append_instead_of_extend.rs

```diff
@@ -0,0 +1,82 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::source::snippet_with_applicability;
+use clippy_utils::ty::{is_type_diagnostic_item, is_type_lang_item};
+use if_chain::if_chain;
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind, LangItem};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::symbol::sym;
+
+declare_clippy_lint! {
+    /// **What it does:** Checks for occurrences where one vector gets extended instead of append
+    ///
+    /// **Why is this bad?** Using `extend` instead of `append` is more precise and faster
+    ///
+    /// **Known problems:** None.
+    ///
+    /// **Example:**
+    ///
+    /// ```rust
+    /// let mut a = vec![1, 2, 3];
+    /// let mut b = vec![4, 5, 6];
+    ///
+    /// // Bad
+    /// a.extend(b.drain(..));
+    ///
+    /// // Good
+    /// a.append(&mut b);
+    /// ```
+    pub APPEND_INSTEAD_OF_EXTEND,
+    perf,
+    "default lint description"
+}
+
+declare_lint_pass!(AppendInsteadOfExtend => [APPEND_INSTEAD_OF_EXTEND]);
+
+impl<'tcx> LateLintPass<'tcx> for AppendInsteadOfExtend {
```

Comment:
I've previously that we've combined all lint passes regarding methods into a single lint pass for performance reasons ([here](https://github.com/rust-lang/rust-clippy/pull/7270#discussion_r638725620)). Could you move this implementation into the `methods` module with the single lint pass as well? Here was the previous rough guide:

> 3. Now the next thing is probably something you couldn't know. Clippy already as a bunch of lints focusing on methods. They have therefore all been combined in a single module and lint pass. New lints focussing on methods should be injected into the existing context. The [`STRING_EXTEND_CHARS`]() is probably a good example for you as it does something very similar. Here are some links:
>    * Injecting into current method lint-pass [-->](https://github.com/rust-lang/rust-clippy/blob/cd241b33cb8379311286f53c4afc6e671ddfded1/clippy_lints/src/methods/mod.rs#L1985)
>    * Lint implementation: [-->](https://github.com/rust-lang/rust-clippy)

Reply:
Done moved lint to method